### PR TITLE
Fixes #4294: Profile successfully deleted message

### LINKED ISSUE #4294: After deleting a profile, Deleted successfully message should be displayed
**Describe the bug**
After deleting a profile, Deleted successfully notification should be displayed 

**To Reproduce**
1. Login to Oppia as Administrator
2. Click on Administrator Controls- Enter PIN
3. Click on Edit Profiles
4. Select a profile to delete - Click on Profile Deletion
5. A confirmation dialog appears- Click on Delete 

**Expected behavior**
After clicking on Delete a message 'Deleted successfully' should be seen and read by talk back. This helps the talk back users to confirm that the profile has been deleted

**Environment**
Device name: One plus Nord2 5G
Android version : Android 11
App version : 0.7-alpha-d1b2827517

### PR DESCRIPTION
**Pull request by mbobiosio**
## Explanation
 - "Fixes [#4294](https://github.com/oppia/oppia-android/issues/4294)

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by BenHenning** [comment]
@mbobiosio is this ready for review? If so, please make sure to assign it to whoever needs to review it (per https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section you should make sure that the 'assignees' list always includes whoever next has to take action on the PR).

If it's not ready for review, please mark it as 'Draft' until it is ready to go into review.

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditFragmentPresenter.kt [hunk 1]
```diff
@@ -101,15 +103,16 @@ class ProfileEditFragmentPresenter @Inject constructor(
         ProfileId.newBuilder().setInternalId(internalProfileId).build(),
         binding.profileEditAllowDownloadSwitch.isChecked
       ).toLiveData().observe(
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditFragmentPresenter.kt:105]: Could probably fit on one line now for a bit cleaner code, e.g.: ``).toLiveData().observe(activity) {``

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @mbobiosio! I took an initial high-level pass. My feedback is:
- Please update the PR description to include details of what your PR contains, and how you approached it. PRs like #4239 are good examples of detailed descriptions, though that PR's description is much longer than what's probably needed here. Make sure to also fill in the 'UI' section per the instructions since this is introducing a new UI.
- We shouldn't be using a toast here. Generally, we probably should be using a snackbar, but since we aren't yet utilizing those yet this should be an alert dialog hosted in its own dialog fragment (as we do for other similar notices in the app).
- New tests should be added for changed functionality to make sure the notice appears with the correct text at the correct time.

File: app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditFragmentPresenter.kt [hunk 2]
```diff
@@ -132,22 +135,25 @@ class ProfileEditFragmentPresenter @Inject constructor(
...
-              fragment.startActivity(intent)
-            } else {
-              val intent = Intent(fragment.requireContext(), ProfileListActivity::class.java)
-              intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
-              fragment.startActivity(intent)
-            }
+        fragment
+      ) {
+        if (it is AsyncResult.Success) {
+          fragment.requireContext().toast(
```
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditFragmentPresenter.kt:141]: @BenHenning For accessibility perspective should we call this Toast as soon as it appears?

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt [hunk 3]
```diff
@@ -116,7 +116,9 @@ class AppVersionActivityTest {
 
   @get:Rule
   val activityTestRule: ActivityTestRule<AppVersionActivity> = ActivityTestRule(
-    AppVersionActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+    AppVersionActivity::class.java, /* initialTouchMode= */
+    true, /* launchActivity= */
+    false
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt:121]: These changes can be un-done.

`AppVersionActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@mbobiosio PTAL Thanks.

**Comment by BenHenning** [comment]
@mbobiosio I'll need to close this since it was force-pushed. Please review https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR (particularly point 4 under https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#making-a-local-code-change-using-the-terminal). We don't allow force-pushing since it messes up comment history and can make code reviewing much more difficult. You can open a new PR with a fresh history, just please make sure not to force push again in the future as it will lead to delays in the review process. Thanks!

**Pull request closed without merging**